plugins/jobs/src/js/pages/JobDetailPage.js

```diff
@@ -108,23 +108,21 @@ class JobDetailPage extends mixin(TabsMixin) {
 
     const { job } = this.props;
 
-    // TODO: wait for https://github.com/dcos/dcos-ui/pull/3029 to be merged and remove this line
-    const breadcrumbJob = { ...job, path: [] };
-
     return (
       <Page>
         <Page.Header
           actions={this.getActions()}
           breadcrumbs={
             <Breadcrumbs
-              states={this.renderBreadcrumbStates(breadcrumbJob)}
-              item={breadcrumbJob}
+              jobPath={job.path}
```

Comment:
How about we pass `job` as the only prop and extract `this.renderBreadcrumbStates(job)` as an internal component into `plugins/jobs/src/js/components/Breadcrumbs.js` because it's a pure function already and uses no state of the `JobDetailPage` class.

That way we would have component composition and separation of responsibility as it doesn't feel like `<JobDetailPage />` component is really responsible for constructing the `<JobInfo />`

So that will become:
```js
<Breadcrumbs job={job} />
```

Or if you decide it's better to still keep `jobInfo`
```js
<Breadcrumbs job={job} jobInfo={<JobInfo job={job} />} />
```

Reply:
we cant pass `job` because overview, error and loading pages doesnt have one. 😕

also we explicitly wanted to remove that extra logic from the breadcrumb component and allow whatever component uses it to pass in jsx which shall be rendered in the `infofield`.

reason is, and thats IMHO, it is bad practice to pass in such a huge thing and do "magic" on it instead of just pass in what is needed and to as less logic as possible in the component.

Reply:
we could replace `renderBreadcrumbStates` with a `<JobInfo>` component, but that isnt in scope for this PR.